File: SecondYear/aayushker_AayushkerSingh_2226CSE(AI)1010_2/Week_3/Sorting_the _Sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        w = s.split()
    
        original_order = [''] * len(w)
        
        for word in w:
            last_char = word[-1]

            if last_char.isdigit():
                index = int(last_char)
                original_order[index - 1] = word[:-1]

        return ' '.join(original_order)

File: SecondYear/any-mesh_AnimeshKumarSingh_2226CSE(AI&ML)1119_2/Week_3/sortingTheSentence.cpp
class Solution {
public:
    string sortSentence(string s) {
        string ans;
        vector<pair<int, string>> p;
        istringstream ss(s);
        string word;
        while(ss >> word){
            int index = word[word.length() - 1];
            word.pop_back();
            p.push_back({index, word});
        }

        sort(p.begin(), p.end());

        for(auto x : p){
            ans += x.second;
            ans += " ";
        }
        ans.pop_back();
        return ans;
    }
};
